components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -432,7 +452,11 @@ func (g *RTKMovementSensor) ReceiveAndWriteI2C(ctx context.Context) {
 
 		msg, err := scanner.NextMessage()
 		if err != nil {
-			g.ntripStatus = false
```

Comment:
I think it's cleaner here (and in the other places this is done) to just call unlock normally after the property has been set, rather than initialize and call an anonymous function

Reply:
I don't feel strongly here, and am happy to change this up. If the assignment was something that could fail (e.g., `g.ntripStatus = complicatedFunction()`, I'd feel more strongly about unlocking the mutex with a `defer` statement: if something panics with the mutex locked, it's possible that we recover from the problem elsewhere but deadlock because the mutex never gets unlocked. What I'm really trying to do is RAII, where you lock the mutex when you enter a certain scope and unlock it again when you leave that scope.

but you're right that with such a simple piece of code, just calling Lock() and Unlock() is easier to read.

Reply:
Gautham and I discussed: we're on the same page here. I've left alone the one place that might fail with the mutex locked, and changed the others to be more readable.